### PR TITLE
YupYup Enemies Mostly Good

### PR DESCRIPTION
**Pull request by Jeff-Moreau**

### CODE REVIEW

File: Source/SteveSloth/Private/MyEnemyStateComponent.cpp [hunk 1]
```diff
@@ -9,6 +9,7 @@
  ****************************************************************************************/
 
 #include "MyEnemyStateComponent.h"
+#include "MyEnemyBaseClass.h"
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyEnemyStateComponent.cpp:12]: are we using this still

File: Source/SteveSloth/Private/MyGenericEnemyIdleState.cpp [hunk 2]
```diff
@@ -22,9 +23,12 @@ void UMyGenericEnemyIdleState::ExitState()
 
 void UMyGenericEnemyIdleState::UpdateState(float deltaTime)
 {
+	UE_LOG(LogTemp, Warning, TEXT("Update Idle"));
```
Comment (MysticAmazon) [inline Source/SteveSloth/Private/MyGenericEnemyIdleState.cpp:26]: forgot the ue log.

File: Source/SteveSloth/Private/MyGenericEnemyIdleState.cpp [hunk 3]
```diff
@@ -22,9 +23,12 @@ void UMyGenericEnemyIdleState::ExitState()
...
+
 	// Play the Animation for being Idle
 	if (Myself != nullptr && !IsAnimationRunning)
 	{
+		UE_LOG(LogTemp, Warning, TEXT("Animation %s"), *Myself->GetName());
```
Comment (MysticAmazon) [inline Source/SteveSloth/Private/MyGenericEnemyIdleState.cpp:31]: also this one

File: Source/SteveSloth/Private/MyGenericEnemyIdleState.cpp [hunk 4]
```diff
@@ -12,6 +12,7 @@
 
 void UMyGenericEnemyIdleState::EnterState()
 {
+	UE_LOG(LogTemp, Warning, TEXT("Enter Idle"));
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyGenericEnemyIdleState.cpp]: remove

File: Source/SteveSloth/Private/MyGenericEnemyIdleState.cpp (hunk 2 above)
Comment (Saretzky) [inline Source/SteveSloth/Private/MyGenericEnemyIdleState.cpp]: remove

File: Source/SteveSloth/Private/MyGenericEnemyIdleState.cpp (hunk 3 above)
Comment (Saretzky) [inline Source/SteveSloth/Private/MyGenericEnemyIdleState.cpp]: remove

File: Source/SteveSloth/Private/MyGenericEnemyRangeAttackState.cpp [hunk 5]
```diff
@@ -45,5 +46,15 @@ void UMyGenericEnemyRangeAttackState::SetEnemyMesh(USkeletalMeshComponent* mesh)
 
 void UMyGenericEnemyRangeAttackState::LaunchProjectile()
 {
-	// Instantiate certain projectile here just need actor and location
+	UWorld* myWorld = Myself->GetMyWorld();
+	FTransform myTransform = FTransform(Myself->GetActorTransform());
+	TSubclassOf<AActor> projectileSpawn = Myself->GetProjectile();
+
+	UE_LOG(LogTemp, Warning, TEXT("Throw"));
```
Comment (MysticAmazon) [inline Source/SteveSloth/Private/MyGenericEnemyRangeAttackState.cpp:53]: this one to

File: Source/SteveSloth/Private/MyGenericEnemyRangeAttackState.cpp [hunk 6]
```diff
@@ -45,5 +46,15 @@ void UMyGenericEnemyRangeAttackState::SetEnemyMesh(USkeletalMeshComponent* mesh)
...
+	
+	if (myWorld && projectileSpawn)
+	{
+		UE_LOG(LogTemp, Warning, TEXT("Not Null"));
```
Comment (MysticAmazon) [inline Source/SteveSloth/Private/MyGenericEnemyRangeAttackState.cpp:57]: and this one

File: Source/SteveSloth/Private/MyKoalaEnemyClass.cpp [hunk 7]
```diff
@@ -38,11 +97,12 @@ void AMyKoalaEnemyClass::Tick(float DeltaTime)
...
 	//Switching States:
 	//if the enemy is in ranged distance
-	if (isHit && outHit.bBlockingHit && outHit.GetActor() == Player && !bIsAttackingRanged)
+	//if (/*isHit && outHit.bBlockingHit && */outHit.GetActor() == Player && !bIsAttackingRanged)
+	if (FVector::Dist(this->GetActorLocation(), Player->GetActorLocation()) <= RangedAttackRange && FVector::Dist(this->GetActorLocation(), Player->GetActorLocation()) > ChaseRange && !bIsAttackingRanged)
 	{
 		StateMachine->ChangeState(StateMachine->GetState(RangedAttack));
 		bIsAttackingRanged = true;
-
+		UE_LOG(LogTemp, Warning, TEXT("In Ranged Attack State"));
```
Comment (MysticAmazon) [inline Source/SteveSloth/Private/MyKoalaEnemyClass.cpp:105]: this one to

File: Source/SteveSloth/Private/MyKoalaEnemyClass.cpp [hunk 8]
```diff
@@ -67,19 +127,21 @@ void AMyKoalaEnemyClass::Tick(float DeltaTime)
...
 		bIsPatroling = false;
 		bIsAttackingRanged = false;
 	}
-	else if (FVector::Dist(this->GetActorLocation(), StartingLocation.GetLocation()) <= PatrolRange && !bIsPatroling)
+	else if (FVector::Dist(this->GetActorLocation(), StartingLocation.GetLocation()) <= PatrolRange && !bIsPatroling && !bIsChasing)
 	{
 		StateMachine->ChangeState(StateMachine->GetState(Patrol));
-		bIsPatroling = true;
 
+		UE_LOG(LogTemp, Warning, TEXT("In Patrol State"));
```
Comment (MysticAmazon) [inline Source/SteveSloth/Private/MyKoalaEnemyClass.cpp:142]: also here

**Review by MysticAmazon — CHANGES_REQUESTED** [review]

File: Source/SteveSloth/Private/MyGenericEnemyRangeAttackState.cpp (hunk 5 above)
Comment (Saretzky) [inline Source/SteveSloth/Private/MyGenericEnemyRangeAttackState.cpp]: remove

File: Source/SteveSloth/Private/MyGenericEnemyRangeAttackState.cpp (hunk 6 above)
Comment (Saretzky) [inline Source/SteveSloth/Private/MyGenericEnemyRangeAttackState.cpp]: remove

File: Source/SteveSloth/Private/MyKoalaEnemyClass.cpp (hunk 7 above)
Comment (Saretzky) [inline Source/SteveSloth/Private/MyKoalaEnemyClass.cpp:105]: remove

File: Source/SteveSloth/Private/MyKoalaEnemyClass.cpp
Comment (Saretzky) [inline Source/SteveSloth/Private/MyKoalaEnemyClass.cpp:1]: you're gonna have to do it for all the enemies right?

File: Source/SteveSloth/Private/MyEnemyStateComponent.cpp [hunk 9]
```diff
@@ -18,25 +19,25 @@ UMyEnemyStateComponent::UMyEnemyStateComponent()
 void UMyEnemyStateComponent::BeginPlay()
 {
 	Super::BeginPlay();
+
```
Comment (CrazyTobi) [inline Source/SteveSloth/Private/MyEnemyStateComponent.cpp:22]: Forgot to delete this white space

File: Source/SteveSloth/Private/MyKoalaEnemyClass.cpp (hunk 8 above)
Comment (Saretzky) [inline Source/SteveSloth/Private/MyKoalaEnemyClass.cpp:142]: remove

**Review by CrazyTobi — CHANGES_REQUESTED** [review]

File: Source/SteveSloth/Private/MyKoalaEnemyClass.cpp [hunk 10]
```diff
@@ -96,6 +158,7 @@ void AMyKoalaEnemyClass::Tick(float DeltaTime)
 void AMyKoalaEnemyClass::StartFleeState()
 {
 	StateMachine->ChangeState(StateMachine->GetState(Flee));
+	UE_LOG(LogTemp, Warning, TEXT("In Flee State"));
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyKoalaEnemyClass.cpp:161]: remove

File: Source/SteveSloth/Private/MyKoalaEnemyClass.cpp [hunk 11]
```diff
@@ -109,4 +172,18 @@ void AMyKoalaEnemyClass::RangedAttackOver()
 	bIsIdle = false;
 	bIsChasing = false;
 	bIsPatroling = false;
+}
+
+void AMyKoalaEnemyClass::StartPatrol()
+{
+	StateMachine->ChangeState(StateMachine->GetState(Patrol));
+
+	UE_LOG(LogTemp, Warning, TEXT("In Patrol State"));
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyKoalaEnemyClass.cpp:181]: remove

File: Source/SteveSloth/Private/MyProjectileBaseClass.cpp [hunk 12]
```diff
@@ -30,7 +30,7 @@ AMyProjectileBaseClass::AMyProjectileBaseClass()
 void AMyProjectileBaseClass::BeginPlay()
 {
 	Super::BeginPlay();
-
+	UE_LOG(LogTemp, Warning, TEXT("Projectile Here"));
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyProjectileBaseClass.cpp:33]: remove

**Review by Saretzky — CHANGES_REQUESTED** [review]

File: Source/SteveSloth/Private/MyKoalaEnemyClass.cpp
Comment (Saretzky) [reply Source/SteveSloth/Private/MyKoalaEnemyClass.cpp:1]: all the state stuff i mean

**Review by Saretzky — APPROVED** [review]
looks good

**Review by MysticAmazon — APPROVED** [review]
everything looks good !

**Review by CrazyTobi — APPROVED** [review]
Looks good

**Pull request merged**